llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -110,13 +110,13 @@ static uint64_t getObjectSize(const Value *V, const DataLayout &DL,
   Opts.RoundToAlign = RoundToAlign;
   Opts.NullIsUnknownSize = NullIsValidLoc;
   if (getObjectSize(V, Size, DL, &TLI, Opts))
-    return Size;
-  return MemoryLocation::UnknownSize;
+    return LocationSize(Size, DL.getTypeAllocSize(V->getType()).isScalable());
```

Comment:
`getObjectSize()` doesn't support scalable vectors, so this change doesn't really make sense to me (same in DSE).

If it's helpful using code to return LocationSize here that's fine, but we shouldn't use this misleading `isScalable()` check here.

Reply:
I see, ok